sklearn/tests/test_discriminant_analysis.py

```diff
@@ -525,23 +541,32 @@ def test_qda_store_covariance():
 def test_qda_regularization():
     # the default is reg_param=0. and will cause issues
     # when there is a constant variable
+    # UserWarning will raised due to the constant variable
+    # Runtime warning will be raised due to unregularized
+    # constant variable in the covariance matrix.
     clf = QuadraticDiscriminantAnalysis()
-    with ignore_warnings():
-        y_pred = clf.fit(X2, y6).predict(X2)
+    with (pytest.warns(UserWarning) and
+          pytest.warns(RuntimeWarning)):
+        y_pred = clf.fit(X2, y6)
+        y_pred = clf.predict(X2)
     assert np.any(y_pred != y6)
 
     # adding a little regularization fixes the problem
+    # No Runtime Warning will be raised with regularization
+    # But UserWarning will persist
     clf = QuadraticDiscriminantAnalysis(reg_param=0.01)
-    with ignore_warnings():
+    with pytest.warns(UserWarning):
         clf.fit(X2, y6)
-    y_pred = clf.predict(X2)
+        y_pred = clf.predict(X2)
```

Comment:
Is it the fit or the predict that raises the warning? Or both?

Reply:
Just fit has the warning. Predict doesn't raise it. Fixed it to show it consistently.